server.js

```diff
@@ -12,16 +12,105 @@ mongoose.Promise = Promise;
 const port = process.env.PORT || 8080;
 const app = express();
 
+const listEndpoints = require('express-list-endpoints');
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// SCHEMA GOES HERE
+const { Schema } = mongoose;
+
+const ThoughtSchema = new Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  heart: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: () => Date.now()
+  }
+});
+
+const Thought = mongoose.model('Thought', ThoughtSchema)
+
+// ROUTES GOES HERE
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.status(200).send({
+    success: true,
+    message: "OK",
+    body: {
+      content: "Welcome to Happy Thoughts API!",
+      endpoints: listEndpoints(app)
+    }
+  });
+});
+
+// GET 20 LATEST THOUGHTS
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc', _id: 'desc' }).limit(20).exec();
+  res.status(200).json(thoughts);
 });
 
+// GET THOUGHT BY ID
+app.get("/thoughts/id/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const singleThought = await Thought.findById(id)
+    if (singleThought) {
+      res.status(200).json(singleThought);
+    } else {
+      res.status(400).json('Not found');
+    }
+    } catch (e) {
+      res.status(400).json('No such id');
+    }
+});
+
+app.post("/thoughts", async (req, res) => {
+  const { message } = req.body;
+  try {
+    const thought = await new Thought({ message }).save();
+    res.status(201).json({
+      success: true,
+      response: thought,
+      message: 'Happy thought successfully sent!'
+    })
+  } catch (e) {
+    res.status(400).json({
+      success: false,
+      response: e,
+      message: 'Error! Happy thought NOT sent successfully!'
+    })
+  }
+});
+
+// PATCH
+app.patch("/thoughts/:_id/like", async (req, res) => {
```

Comment:
I used patch first as well but then I saw in the instructions that they wanted a post. I don't know which one is most accurate in this case, just a reflection.